vcx/libvcx/src/messages/get_message.rs

```diff
@@ -331,6 +332,7 @@ impl Message {
                 } else {
                     // POTENTIAL BUG: we are losing the thread that was potentially captured
                     // from what was decrypted!
+                    new_message.decrypted_payload = Some(data);
```

Comment:
I think this was my bug. I wasn't saving decrypted_payload when V2 decryption had been done.